### PR TITLE
Add metube extension

### PR DESCRIPTION
**Pull request by dotvhs**
## Description

This extension communicates with [MeTube](https://github.com/alexta69/metube) and allows for adding links to download quicker. It implements two commands, one that sends clipboard to MeTube and one that allows adding through a form if person needs to change Quality or Format of media being downloaded.

## Screencast

https://github.com/user-attachments/assets/5dcf4299-3f1c-45c0-9b55-d93357250e18

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/metube/src/add-to-metube.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  url: string;
+  quality: string;
+  format: string;
+};
+
+interface Preferences {
+  metubeUrl: string;
+  quality: string;
+  format: string;
+}
```
Comment (pernielsentikaer) [inline extensions/metube/src/add-to-metube.tsx:15]:
```suggestion
```
This is already a part of `raycast-env.d.ts`

File: extensions/metube/src/add-to-metube.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+export default function Command() {
+  const preferences = getPreferenceValues<Preferences>();
+  const quality = preferences.quality;
+  const format = preferences.format;
+
+  async function handleSubmit(values: Values) {
+    if (!values.url) {
```
Comment (pernielsentikaer) [inline extensions/metube/src/add-to-metube.tsx:23]: You can use [useForm](https://developers.raycast.com/utilities/react-hooks/useform) which has validation, errors and handles states

File: extensions/metube/src/clipboard-to-metube.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { showHUD, Clipboard } from "@raycast/api";
+import { addToQueue } from "./helper.js";
+
+export default async function main() {
+  const clip = await Clipboard.readText();
+  if (clip) {
```
Comment (pernielsentikaer) [inline extensions/metube/src/clipboard-to-metube.ts:6]: Here you can also validate if it's even a link (and maybe if it's a youtube?)

File: extensions/metube/src/helper.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+interface AddToQueueResponse {
+  status: string;
+  msg?: string;
+}
+
+interface Preferences {
+  metubeUrl: string;
+  quality: string;
+  format: string;
+}
```
Comment (pernielsentikaer) [inline extensions/metube/src/helper.ts:13]:
```suggestion
```
This is already a part of `raycast-env.d.ts`

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/metube/src/add-to-metube.tsx (hunk 2 above)
Comment (dotvhs) [reply extensions/metube/src/add-to-metube.tsx:23]: Oh! I see, thank you, I was not aware of this, looks interesting, I will implement this for sure.

File: extensions/metube/src/clipboard-to-metube.ts (hunk 3 above)
Comment (dotvhs) [reply extensions/metube/src/clipboard-to-metube.ts:6]: I could validate if it follows a generic link pattern, sure. However, MeTube works through yt-dlp which supports lots and lots of websites and not just YT, so I don't think limiting it to just YT would be helpful in this case but I still appreciate the suggestion.

**Comment by dotvhs** [comment]
I finished requested changes. I added verification, removed the unnecessary parts of code and I'm checking if content is a link.

Hopefully I did things correct this time, thank you for your time.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**